core/extensions.rs

```diff
@@ -2,11 +2,22 @@
 use crate::OpState;
 use anyhow::Error;
 use std::cell::RefCell;
+use std::path::PathBuf;
 use std::rc::Rc;
 use std::task::Context;
 use v8::fast_api::FastFunction;
 
-pub type SourcePair = (&'static str, &'static str);
+pub enum ExtensionSourceFileSource {
+  /// Use this option when snapshotting so that the source code is not embedded
+  /// in the produced binary.
+  File(PathBuf),
+  /// Use this option when not snapshotting.
+  Embedded(String),
```

Comment:
Why a heap allocated String? It should be `&'static str`

Reply:
You can also get rid of `.to_string()` in the usage places

Reply:
Fixed